source/mongodb-handshake/handshake.rst

```diff
@@ -224,20 +224,57 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
-Limitations
+Supporting Wrapping Libraries
 ===========
 
-The entire metadata BSON document MUST NOT exceed 512 bytes. This includes all
-BSON overhead.  The ``client.application.name`` cannot exceed 128 bytes.  MongoDB
-will return an error if these limits are not adhered to, which will result in
-handshake failure. Drivers MUST validate these values and truncate driver
-provided values if necessary.
+Drivers MUST allow libraries which wrap the driver to amend the client metadata
```

Comment:
Quoting a basic definition of "amend":

> make minor changes in (a text) in order to make it fairer, more accurate, or more up-to-date.

I think "append to" would be more appropriate, as @behackett clarified in [SPEC-1502](https://jira.mongodb.org/browse/SPEC-1502) that we don't want any higher level libraries to overwrite anything that would normally be set by the driver itself.

Reply:
done